fastlane/Fastfile

```diff
@@ -207,104 +164,19 @@ platform :ios do
-    changelog_path = edit_changelog(generated_contents: generated_contents)
-    changelog = File.read(changelog_path)
-    
-    create_new_release_branch(version: new_version_number)
-    replace_version_number(version: new_version_number)
-
-    attach_changelog_to_master(new_version_number)
-
-    commit_updated_files_and_push(version: new_version_number)    
-    
-    create_pull_request(
-      title: "Release/#{new_version_number}",
-      base: "main",
-      body: changelog
-    )
-  end
-
-  # Inspired by fastlane's changelog
-  # https://github.com/fastlane/fastlane/blob/eb4584e0394c1fecccd70d4313260f4707f2af80/fastlane/Fastfile#L346-L380
-  desc "Generate changelog from GitHub compare and PR data for mentioning GitHub usernames in release notes"
-  lane :github_changelog do |options|
-    last_version = sh("git describe --tags --abbrev=0").strip
-    old_version = options[:old_version] || last_version
-
-    UI.important("Auto-generating changelog since #{old_version}")
-
-    org = "RevenueCat"
-    repo = "purchases-ios"
-
-    path = "/repos/#{org}/#{repo}/compare/#{old_version}...HEAD"
-    api_token = ENV["GITHUB_TOKEN"]
-
-    # Get all commits from previous version (tag) to HEAD
-    resp = github_api(path: path, api_token: api_token)
-    body = JSON.parse(resp[:body])
-    commits = body["commits"].reverse
-
-    # TODO: Fix this process where rate limiting is not an issue
-    # Temporary workaround to get around GitHub rate limit
-    rate_limit_sleep = ENV["GITHUB_RATE_LIMIT_SLEEP"].to_i
-
-    formatted = commits.map do |commit|
-      if rate_limit_sleep > 0
-        UI.message("Sleeping #{rate_limit_sleep} second(s) to avoid rate limit 🐌")
-        sleep(rate_limit_sleep)
-      end
-
-      # Default to commit message info
-      message = commit["commit"]["message"].lines.first.strip
-      name = commit["commit"]["author"]["name"]
-      username = commit["author"]["login"]
-
-      # Get pull request associate with commit message
-      sha = commit["sha"]
-      pr_resp = github_api(path: "/search/issues?q=repo:#{org}/#{repo}+is:pr+base:main+SHA:#{sha}", api_token: api_token)
-      body = JSON.parse(pr_resp[:body])
-      items = body["items"]
-
-      if items.size == 1
-        item = items.first
-        message = "#{item['title']} (##{item['number']})"
-        username = item["user"]["login"]
-      else
-        UI.user_error!("Cannot generate changelog. Multiple commits found for #{sha}")
-      end
-
-      "* #{message} via #{name} (@#{username})"
-    end.join("\n")
-
-    puts formatted if options[:print]
-
-    formatted
+    begin
+      ensure_git_status_clean
+    rescue => e
+      UI.error("RevenueCat-Swift.h autogenerated file changed. Make sure it's up-to-date before releasing.")
+    end
   end
 
   desc "Creates RevenueCat-Swift.h for a new release"
-  private_lane :compile_autogenerated_header do |options|
+  lane :compile_autogenerated_header do |options|
```

Comment:
I made this public in case a dev wants to recreate `RevenueCat-Swift.h` header, which might be useful if we now catch this as an error during the `release-checks` lane.

Reply:
👍